lib/ibmiconnector.js

```diff
@@ -645,34 +662,78 @@ class IBMiConnector extends SQLConnector {
    * @returns {ParameterizedSQL|*} Database column value.
    *
    */
-  toColumnValue(propertyDef, value) {
-    debug('IBMiConnector.prototype.toColumnValue prop=%j value=%j', propertyDef, value);
-    if (value === null) {
-      if (propertyDef.autoIncrement || propertyDef.id) {
-        return new ParameterizedSQL('DEFAULT');
-      }
+  toColumnValue(prop, val) {
+    if (val === undefined && this.isNullable(prop)) {
       return null;
     }
-    if (!propertyDef) {
-      return value;
+    if (val === null) {
+      if (this.isNullable(prop)) {
+        return val;
+      } else {
```

Comment:
Nitpick: Since you are returning in the `if` branch, this `else` branch is not needed. Without `else`, you can avoid one level of nesting/indentation.

```js
if (this.isNullable(prop)) {
  return val;
}

try {
 // ...
} catch (err) {
  // ...
}
```